packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -51,6 +53,20 @@ import type {
   UseTRPCSuspenseQueryResult,
 } from './types';
 
+const trackResult = <T extends object>(
```

Comment:
if connection is a prop, we need this to track keys of that prop

e.g. `onTrackedResult(['connection'], ['error')`